### PR TITLE
Functional Tests : Migration TS (FO > 09)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : Migration TS (FO > 09)
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #30263
| Related PRs       | N/A
| How to test?      | Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/3684334602
| Possible impacts? | N/A

### CODE REVIEW

File: tests/UI/pages/FO/product/index.ts [hunk 1]
```diff
@@ -1,13 +1,128 @@
...
+
+import type {Page} from 'playwright';
 
 /**
  * Product page, contains functions that can be used on the page
  * @class
  * @extends FOBasePage
  */
-class Product extends FOBasePage {
+export default class Product extends FOBasePage {
```
Comment (NeOMakinG) [inline tests/UI/pages/FO/product/index.ts]:
```suggestion
class Product extends FOBasePage {
```

File: tests/UI/pages/FO/product/index.ts [hunk 2]
```diff
@@ -465,9 +580,7 @@ class Product extends FOBasePage {
    * @param row {Number} the review number in the list
    * @returns {Promise<number>}
    */
-  getReviewRating(page, row = 1) {
+  getReviewRating(page: Page, row: number = 1): Promise<number> {
     return page.$$eval(this.productRatingStar(row), (divs) => divs.length);
   }
 }
-
-module.exports = new Product();
```
Comment (NeOMakinG) [inline tests/UI/pages/FO/product/index.ts:473]: `export default new Product();`

**Review by boubkerbribri — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

File: tests/UI/pages/FO/product/index.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/pages/FO/product/index.ts]: Fixed

File: tests/UI/pages/FO/product/index.ts (hunk 2 above)
Comment (Progi1984) [reply tests/UI/pages/FO/product/index.ts:473]: Fixed

File: tests/UI/campaigns/functional/FO/09_productPage/02_addToCart.ts [hunk 3]
```diff
@@ -185,8 +185,8 @@ describe('FO - product page : Add product to cart', async () => {
 
         // Create products images
         await Promise.all([
-          files.generateImage(productData.coverImage),
-          files.generateImage(productData.thumbImage),
+          files.generateImage(productData.coverImage ?? ''),
+          files.generateImage(productData.thumbImage ?? ''),
```
Comment (NeOMakinG) [inline tests/UI/campaigns/functional/FO/09_productPage/02_addToCart.ts:189]: Why `productData.coverImage` could be undefined? It looks like it's always defined

**Review by NeOMakinG — APPROVED** [review]
A small comment but lgtm 🍬

File: tests/UI/campaigns/functional/FO/09_productPage/02_addToCart.ts (hunk 3 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/FO/09_productPage/02_addToCart.ts:189]: Because it's not always defined with `Product` type.

**Comment by boubkerbribri** [comment]
Thanks @Progi1984

**Pull request merged**